### PR TITLE
[pipeline/trusted-resources]Verify task

### PR DESCRIPTION
**Pull request by Yongxuanzhang**
# Changes

This commit will introduce the validation of task, after verifying the
taskrun, the refered task will also be fetched and verified.
The second change is that webhook will make a copy of object and exclude
system populated fields. And verify on the copied object only on create
operation.

Before this commit, the webhook can only verify on taskspec.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Commit messages includes a project tag in the subject - e.g. [OCI], [hub], [results], [task-loops]

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: pipeline/trusted-resources/cmd/sign/sign_test.go [hunk 1]
```diff
@@ -178,9 +135,49 @@ func unmarshal(t *testing.T, buf []byte) (*v1beta1.TaskRun, []byte) {
 	if err != nil {
 		t.Fatalf("error decoding signature: %v", err)
 	}
+
+	delete(tr.ObjectMeta.Annotations, trustedtask.SignatureAnnotation)
```
Comment (wlynch) [inline pipeline/trusted-resources/cmd/sign/sign_test.go:139]: Why do we need to delete the annotation here? (wouldn't it already be removed prior to signing?)

**Comment by Yongxuanzhang** [comment]
will divide this pr

File: pipeline/trusted-resources/cmd/sign/main.go [hunk 2]
```diff
@@ -119,3 +118,25 @@ func Sign(ctx context.Context, tr *v1beta1.TaskRun, signer sigstore.Signer, writ
...
+	}
+
+	signedBuf, err := yaml.Marshal(task)
+	if err != nil {
+		return err
+	}
+
+	_, err = writer.Write(signedBuf)
+	return err
+}
```
Comment (wlynch) [inline pipeline/trusted-resources/cmd/sign/main.go:142]: You should be able to combine these by using the [object interface](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Object)

File: pipeline/trusted-resources/pkg/trustedtask/verify.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	if annotations[kmsAnnotation] != "" {
+		// Fetch key from kms.
+		return kms.Get(ctx, annotations[kmsAnnotation], crypto.SHA256)
+	} else {
+		cosignPublicKeypath := secretPath
+		// Overwrite the path if set in configmap.
+		cm, err := k8sclient.CoreV1().ConfigMaps(system.Namespace()).Get(ctx, signingConfigMap, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
```
Comment (wlynch) [inline pipeline/trusted-resources/pkg/trustedtask/verify.go:49]: This snuck in the last PR - but we'll want to move this out of the request path. We generally shouldn't need to fetch this per request - we can update the reconciler state when the config map is modified. @chuangw6 can help you get this set up for the reconciler.

**Pull request closed without merging**